### PR TITLE
Fix CI issues: Add page card detached from the DOM error + Samples timeout (GitHub rate limite issue) + htpasswd identity provider error (i18n missing key)

### PR DESCRIPTION
**Pull request by jerolimov**
Why does it include multiple fixes instead of fixing one after another? Because the failure rate is so height that its too annoying to wait for 10 reruns to get a single fix in. :man_shrugging: 

If this adds new issues, I kept separate commits to make it possible to find and separate individual changesets.

----

<details>
<summary>Commit 1: OCPBUGS-7195: Add page/import from git/samples cypress timeout (incl. a e2e test refactoring)</summary>

**Related issues:**

* [OCPBUGS-7195](https://issues.redhat.com/browse/OCPBUGS-7195) [CI-Watcher] e2e issue with tests: Create Samples Page Timeout Error
* [Search shows this is fixed for samples](https://search.ci.openshift.org/?search=data-test%3D%22item&maxAge=336h&context=1&type=junit&name=pull-ci-openshift-console-master-e2e-gcp-console&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job) already closed by @sanketpathak in #12574 :+1:

**Analytics/Root cause:**

I started with this work before I noticed that @sanketpathak has done something similar PR #12574.

The root cause is that the **Add page re-renders** the items **multiple** times. If a card is shown or not depends on feature flags and other checks so that appear one after another. It goes so quickly that it isn't a big issue for the user. 

But because the columns were rendered like this, the cards are removed and added into a another column if a card with a higher priority is added:

```yaml
<div>
  <div>{cardsForColumn1}</div>
  <div>{cardsForColumn2}</div>
  <div>{cardsForColumn3}</div>
  <div />
</div>
```

With keeping this implementation it can happen to all cards here that the dom node was moved (removed and added) and that cypress fails with the error:

```
{Timed out retrying after 40050ms: `cy.click()` failed because this element is detached from the DOM.
```

So when looking into this code we should maybe add ({ force: true }) to all the conditions.

https://github.com/openshift/console/blob/713a3e41f33e22d4ac91f9d1fe57689ee89a563a/frontend/packages/dev-console/integration-tests/support/pages/add-flow/add-page.ts#L10-L137

**Fix:** Use click({ `force: true` }) for all add card clicks. But instead of continuing with the big switch statement that sometimes checks the `enum addOptions` and also accepts a string, I changed the code to an implementation that lookup the card ID from a page object.
</details>

<details>
<summary>Commit 2: OCPBUGS-7195: Another sample cypress timeout -> Add GitHub API mocks when importing a Devfile sample</summary>

**Related issues:**

* [OCPBUGS-7195](https://issues.redhat.com/browse/OCPBUGS-7195) [CI-Watcher] e2e issue with tests: Create Samples Page Timeout Error, error locks similar to the first issue but has another root cause 
* [Search for `Timed out retrying after 40000ms: expected '<button.pf-c-button.pf-m-primary.pf-m-disabled>' to be 'enabled'`](https://search.ci.openshift.org/?search=Timed+out+retrying+after+40000ms%3A+expected+%27%3Cbutton.pf-c-button.pf-m-primary.pf-m-disabled%3E%27+to+be+%27enabled%27&maxAge=336h&context=1&type=junit&name=pull-ci-openshift-console-master-e2e-gcp-console&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job) over the last 14 days: 434 runs, 77% failed, 3% of failures match = 3% impact

**Analytics / Root cause**:

When running the tests again and again (locally or on the same CI host?) some tests fails because we hit a GitHub rate limit. We had this issue earlier and mocked the GitHub API for this.

**Fix:** This change mocks now also the GitHub API also when importing the Devfile in `add-flow-ci.feature`.

And instead of mocking really specific endpoints...
</details>

<details>
<summary>Commit 3: OCPBUGS-5940: htpasswd identity provider error -> Missing i18n key -> Do not translate CRD extensions before the i18n files are loaded</summary>

**Related issues:**

* [OCPBUGS-5940](https://issues.redhat.com/browse/OCPBUGS-5940) [CI Watcher]: logs in as 'test' user via htpasswd identity provider: Auth test logs in as 'test' user via htpasswd identity provider

    The underlying issue is that the "shipwright-plugin" or "pipelines-plugin" or another translation file wasn't loaded ("i18n key missing").

    [CI Search for `Auth test logs in as`](https://search.ci.openshift.org/?search=Auth+test+logs+in+as&maxAge=336h&context=1&type=junit&name=pull-ci-openshift-console-master-e2e-gcp-console&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job) in the last 14 days (March 1): 433 runs, 77% failed, 7% of failures match = 6% impact

**Analytics / Root cause**: `getResources` calls `getModelExtensionMetadata` which calls `i18n.t` to translate loaded extensions before the translation files were fetched.

https://github.com/openshift/console/blob/04c482a820f3ea8ba43c7db9d79253c28c7662f3/frontend/public/module/k8s/get-resources.ts#L99-L108

This change was introduced a long time ago as part of https://github.com/openshift/console/pull/10330. It's a race condition between loading the extensions and translation files. For an unknown reason the `missingKeyHandler` wasn't called with i18next v19. Starting with v21 I and the PR https://github.com/openshift/console/pull/12124 I could notice these errors. But the errors were logged by i18next as warning also before.

It is possible to enforce the i18n missing key error by adding a 1s sleep into the i18n handler:

i18n-handler

4.12 / i18next v19 / without PR https://github.com/openshift/console/pull/12124

before-1

4.13 / i18next v21 / with PR https://github.com/openshift/console/pull/12124

missing-i18n

**Fix:** To fix the race condition added a loading promise to i18n.js that resolves when the i18n init finishes loading. Added this as an additional condition to the `Promise.all` check (so that it can run in parallel!) to `getResources`.
</details>

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/support/pages/add-flow/add-page.ts [hunk 1]
```diff
@@ -1,139 +1,40 @@
...
+      // force is required here because the add page moves dom components
+      // from one column to another when the addtional cards are added
+      // after checking the card conditions.
+      .click({ force: true });
+    app.waitForLoad();
+    if (addPageItem.alternativePageTitleSelector) {
+      // sd
+    } else {
+      detailsPage.titleShouldContain(addPageItem.verifyPageTitle);
+    }
```
Comment (jerolimov) [inline frontend/packages/dev-console/integration-tests/support/pages/add-flow/add-page.ts:32]: TODO: missed a check here for:

```tsx
      case 'Pipeline':
      case addOptions.Pipeline:
        cy.wait(3000);
        cy.byTestID('item pipeline').click();
        cy.get('.odc-pipeline-builder-header__title').should(
          'have.text',
          pageTitle.PipelineBuilder,
        );
        app.waitForLoad();
        cy.testA11y(pageTitle.PipelineBuilder);
        break;
```

and

```tsx
      case 'Helm Chart Repositories':
      case addOptions.HelmChartRepositories:
        cy.byTestID('item project-helm-chart-repositories').click();
        cy.get('[data-test="form-title"]').should('have.text', pageTitle.CreateHelmChartRepository);
        cy.testA11y(pageTitle.CreateHelmChartRepository);
        break;
```

and

```tsx
      case 'Create Serverless function':
      case addOptions.CreateServerlessFunction:
        cy.byTestID('item create-serverless-function').click();
        app.waitForLoad();
        cy.get('[data-test-id="resource-title"]').should(
          'have.text',
          pageTitle.CreateServerlessFunction,
        );
        cy.testA11y(pageTitle.CreateServerlessFunction);
        break;
```

**Comment by jerolimov** [comment]
😫

/woof
/retest

**Comment by jerolimov** [comment]
Never give up

/retest

**Comment by rhamilto** [comment]
OCPBUGS-5940 is [impacting CI](https://search.ci.openshift.org/?search=Auth+test+logs+in+as+%27test%27+user+via+htpasswd+identity+provider&maxAge=168h&context=1&type=junit&name=pull-ci-openshift-console-master-e2e-gcp-console&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job).  @jerolimov, I understand your desire to get multiple fixes in given the difficulty to get a single fix it, but coupling these fixes together means none of them are getting in since this PR has been sitting for 2 weeks.  Can we either get this PR prioritized or break the fix for OCPBUGS-5940 out in a separate PR? Thanks!

**Comment by rhamilto** [comment]
Hmm.  I see @jerolimov is on PTO next week.  I'll go ahead and pull the fix for OCPBUGS-5940 in to a separate PR.

**Comment by vikram-raj** [comment]
closing this as these changes got merged

/close

**Pull request closed without merging**